### PR TITLE
Edge cases

### PR DESCRIPTION
**Pull request by sienna-kopf**
Add test for edge case for items not linked to a merchant with a discount do not trigger discount apply even if the item quantity is correct. 

Fix hash access mistake in discount applies model method.

Test in development ✅

100 % RSpec coverage.

### CODE REVIEW

File: app/models/cart.rb [hunk 1]
```diff
@@ -57,7 +57,7 @@ def discounts(item)
   end
 
   def discount_applies
-    @contents.any? do |item_id|
+    @contents.any? do |item_id, quantity|
```
Comment (sienna-kopf) [inline app/models/cart.rb:60]: Oops. Tricky mistake, was returning an array of the id and quantity which obvs couldn't be passed into an AR query

File: spec/features/discounts/cart_spec.rb [hunk 2]
```diff
@@ -171,5 +175,41 @@
...
+
+      within "#cart-item-#{@bluepill.id}" do
+        expect(page).to have_link(@bluepill.name)
+        expect(page).to have_css("img[src*='#{@bluepill.image}']")
+        expect(page).to have_link("#{@bluepill.merchant.name}")
+        expect(page).to have_content("$70.00")
+        expect(page).to have_content("10")
+        expect(page).to have_content("$700.00")
+
+        expect(page).to_not have_content("Item price has been adjusted due to discount!")
```
Comment (sienna-kopf) [inline spec/features/discounts/cart_spec.rb:211]: Makes sure even though there is 10 of this item, it is not of the merchant with the discount so the discount doesn't apply.
